CompilerSource/JDI/src/API/AST.cpp

```diff
@@ -568,13 +568,13 @@ namespace jdi
       dec_literal:
       if (is_letter(content[content.length() - 1])) {
         bool is_float = false;
-        char *number = (char*)alloca(content.length());
-        memcpy(number, content.c_str(), content.length() * sizeof(char));
+        size_t num_len = content.length();
+        const char* number = content.c_str();
         // This block will fail if the number is all letters--but we know it isn't.
-        for (char *i = number + content.length() - 1; is_letter(*i); --i) {
+        for (const char *i = number + content.length() - 1; is_letter(*i); --i) {
```

Comment:
I'm glad this substitution worked; can you refactor it, now, to just count backward from `num_len`?

Reply:
It already is though? Why?

Reply:
What I mean, is, the `i - number` line is icky. Its actual result type is `ptrdiff_t`, which we're just casting to an integer. It's better to just count an integer down instead of the char pointer.

Reply:
Done in 8f1d67a24481f320018cfe8ede41f83a31dcd82e